src/commands/env/env-list.js

```diff
@@ -33,36 +47,55 @@ const envList = async (options, command) => {
     return false
   }
 
-  const siteData = await api.getSite({ siteId })
-  const environment = Object.fromEntries(
-    Object.entries(cachedConfig.env)
-      // Omitting general variables to reduce noise.
-      .filter(([, variable]) => variable.sources[0] !== 'general')
-      .map(([key, variable]) => [key, variable.value]),
+  const { env, siteInfo } = cachedConfig
+  const isUsingEnvelope = siteInfo.use_envelope
+  let environment = env
+
+  if (isUsingEnvelope) {
+    environment = await getEnvelopeEnv({ api, context, env, scope, siteInfo })
+  } else if (context !== 'dev' || scope !== 'any') {
+    error(
+      `To specify a context or scope, please run ${chalk.yellowBright(
+        'netlify open:admin',
+      )} and opt in to the new Environment Variables experience`,
```

Comment:
[Sand] Similar comment above - whatever we decide, we should be consistent. I'll suggest the same change here in case that's the one we want to go with

```suggestion
      )} to open the Netlify UI and opt in to the new environment variables experience from Site settings`,
```